src/core/sys/darwin/netinet/in_.d

```diff
@@ -0,0 +1,605 @@
+
+    extern(D) bool IN_CLASSC(in_addr_t i) pure @safe { return (i & 0xe0000000) == 0xc0000000; }
+    enum IN_CLASSC_NET    = 0xffffff00;
+    enum IN_CLASSC_NSHIFT = 8;
+    enum IN_CLASSC_HOST   = 0x000000ff;
+
+    extern(D) bool IN_CLASSD(in_addr_t i) pure @safe { return (i & 0xf0000000) == 0xe0000000; }
+    enum IN_CLASSD_NET    = 0xf0000000;
+    enum IN_CLASSD_NSHIFT = 28;
+    enum IN_CLASSD_HOST   = 0x0fffffff;
+    extern(D) bool IN_MULTICAST(in_addr_t i) pure @safe { return IN_CLASSD(i); }
+
+    // The fact that these are identical looks suspicious (they're not quite
+    // identical on Linux). However, this _is_ how they're defined in FreeBSD
+    // and on OS X. So, while it _might_ be a bug, if it is, it's an upstream
+    // one, and we're compatible with it.
+    extern(D) bool IN_EXPERIMENTAL(in_addr_t i) pure @safe { return (i & 0xf0000000) == 0xf0000000; }
+    extern(D) bool IN_BADCLASS(in_addr_t i) pure @safe { return (i & 0xf0000000) == 0xf0000000; }
+
+    enum INADDR_UNSPEC_GROUP    = 0xe0000000;
+    enum INADDR_ALLHOSTS_GROUP  = 0xe0000001;
+    enum INADDR_ALLRTRS_GROUP   = 0xe0000002;
+    enum INADDR_ALLRPTS_GROUP   = 0xe0000016;
+    enum INADDR_CARP_GROUP      = 0xe0000012;
+    enum INADDR_PFSYNC_GROUP    = 0xe00000f0;
+    enum INADDR_ALLMDNS_GROUP   = 0xe00000fb;
+    enum INADDR_MAX_LOCAL_GROUP = 0xe00000ff;
+
+    enum IN_LINKLOCALNETNUM = 0xA9FE0000;
+    extern(D) bool IN_LINKLOCAL(in_addr_t i) pure @safe { return (i & IN_CLASSB_NET) == IN_LINKLOCALNETNUM; }
+    extern(D) bool IN_LOOPBACK(in_addr_t i) pure @safe { return (i & 0xff000000) == 0x7f000000; }
+    extern(D) bool IN_ZERONET(in_addr_t i) pure @safe { return (i & 0xff000000) == 0; }
+
+    extern(D) bool IN_PRIVATE(in_addr_t i) pure @safe
+    {
+        return (i & 0xff000000) == 0x0a000000 ||
+               (i & 0xfff00000) == 0xac100000 ||
+               (i & 0xffff0000) == 0xc0a80000;
+    }
+
+    extern(D) bool IN_LOCAL_GROUP(in_addr_t i) pure @safe { return (i & 0xffffff00) == 0xe0000000; }
+
+    extern(D) bool IN_ANY_LOCAL(in_addr_t i) pure @safe { return IN_LINKLOCAL(i) || IN_LOCAL_GROUP(i); }
+
+    enum IN_LOOPBACKNET = 127;
+
+
+    struct ip_opts
+    {
+        in_addr  ip_dst;
+        char[40] ip_opts;
+    };
+
+    enum IP_OPTIONS         = 1;
+    enum IP_HDRINCL         = 2;
+    enum IP_TOS             = 3;
+    enum IP_TTL             = 4;
+    enum IP_RECVOPTS        = 5;
+    enum IP_RECVRETOPTS     = 6;
+    enum IP_RECVDSTADDR     = 7;
+    enum IP_RETOPTS         = 8;
+    enum IP_MULTICAST_IF    = 9;
+    enum IP_MULTICAST_TTL   = 10;
+    enum IP_MULTICAST_LOOP  = 11;
+    enum IP_ADD_MEMBERSHIP  = 12;
+    enum IP_DROP_MEMBERSHIP = 13;
+    enum IP_MULTICAST_VIF   = 14;
+    enum IP_RSVP_ON         = 15;
+    enum IP_RSVP_OFF        = 16;
+    enum IP_RSVP_VIF_ON     = 17;
+    enum IP_RSVP_VIF_OFF    = 18;
+    enum IP_PORTRANGE       = 19;
+    enum IP_RECVIF          = 20;
+
+    enum IP_IPSEC_POLICY = 21;
```

Comment:
Yes, because it says it's deprecated. I saw no point in including declarations where the header file said that they were deprecated or obsolete. A number of the declarations in in.h were marked that way, and some were marked that way on one system, so I left them out on all of the systems. I don't think that anything that's missing is not missing on purpose, and everything that you've mentioned as missing in a comment is purposefully missing. More declarations can be added later if we need to, whereas removing them later is more of a pain. I don't expect that we'll need the ones that I left out.